src/api/libcellml/logger.h

```diff
@@ -146,6 +146,20 @@ class LIBCELLML_EXPORT Logger
      */
     void addIssue(const IssuePtr &issue);
 
+    /**
+     * @brief Remove issue of level ERROR at the specified @p index.
+     *
+     * Removes an error at @p index.  If @c true is returned
+     * then the error at @p index was successfully removed.
+     * If @p index is within the range [0, \#errors)
+     * then @c true is returned, otherwise @c false is returned.
+     *
+     * @param index The index of the error to remove.
+     *
+     * @return @c true if the error is removed, @c false otherwise.
+     */
+    bool removeError(size_t index);
+
```

Comment:
If, internally, we need to call `removeError()`, then I would expect that method to be part of `Logger`'s private implementation, not here in the `protected` section. (Same for `addIssue()` in fact.)

Reply:
Indeed, I think this might be possible with moving `setParent` into a private implementation.  Will look at moving these once that works.

Reply:
Just to be sure: is this going to be done as part of this PR or another one? If the latter, do we have an issue for it?

Reply:
This will be done after [internalise set parent](#876) is done. No issue for it currently.

Reply:
Has now #878.